### PR TITLE
introduce certbot_create_extra_args (for custom RSA key size)

### LINKED ISSUE #100: Feature Request: rsa-key-size
I LOVE this module, its totally perfect, well if I could set the rsa-key-size to 4096 it would be ;-)

I think it should be quite straightforward to implement, we would just need to add this

`--rsa-key-size 4096`

so maybe

`--rsa-key-size {{ rsa_key_size }}`

Many thanks for this module

### PR DESCRIPTION
**Pull request by gagath**
This should fix #100.
It was tested on a real server and I always use it now (on top of #97 PR in my `webroot-enhanced` branch).

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ certbot_certs: []
 certbot_create_command: >-
   {{ certbot_script }} certonly --standalone --noninteractive --agree-tos
   --email {{ cert_item.email | default(certbot_admin_email) }}
+  {{ '--rsa-key-size' if certbot_rsa_key_size else '' }}
+  {{ certbot_rsa_key_size if certbot_rsa_key_size else '' }}
```
Comment (ericsysmin) [inline defaults/main.yml:24]:
```suggestion
  {{ '--rsa-key-size {}'.format(certbot_rsa_key_size) if certbot_rsa_key_size else '' }}
```

Comment (ericsysmin) [inline defaults/main.yml:24]:
```suggestion
{{ (certbot_rsa_key_size is defined) | ternary('--rsa-key-size ~ certbot_rsa_key_size', '')  }}
```

Comment (ericsysmin) [inline defaults/main.yml:24]: May want to try something like this instead, shorter, and on one line
```suggestion
{{ (certbot_rsa_key_size is defined) | ternary('--rsa-key-size ' ~ certbot_rsa_key_size, '')  }}
```

Comment (ericsysmin) [reply defaults/main.yml:24]: @microjoe ^

**Comment by MicroJoe** [comment]
Thanks, updated. I was not able to test it as I do not have any new certificate to generate right now, but I trust you that this will work as fine. No parse error at least.

**Comment by MicroJoe** [comment]
From https://www.jeffgeerling.com/blog/2020/enabling-stale-issue-bot-on-my-github-repositories:

> […] Many issues are opened and never get a follow-up visit from the original author (even if I spend a half hour providing a thorough response). Often this person found a solution elsewhere and never thought to follow-up, or they don't have notifications enabled so they never even know someone responded.

Dear stale bot, we are in the opposite case here: the PR has not been reviewed by the maintainer yet. This PR still applies. Yes, I am negotiating with a bot. :)

**Comment by ericsysmin** [comment]
@geerlingguy i can help if you make me a maintainer

**Comment by simonspa** [comment]
Still love to have this in!

**Comment by geerlingguy** [comment]
I wonder... since there are other CLI options too, which people may want control over... maybe we should just have a var like `certbot_create_extra_args` and you can throw whatever you want in there?

**Comment by MicroJoe** [comment]
Hello geerlingguy, nice to hear from you.

> I wonder... since there are other CLI options too, which people may want control over... maybe we should just have a var like certbot_create_extra_args and you can throw whatever you want in there?

Pros : more flexible, allows to specify a lot of things.
Cons: less discoverable for people not familiar with certbot options.

How about I update this PR to introduce both?

**Comment by simonspa** [comment]
I think this is still relevant. @MicroJoe did you have time to include the suggested `certbot_create_extra_args` as suggested by @geerlingguy ?

**Comment by MicroJoe** [comment]
Commit changed to only introduce `certbot_create_extra_args` so that `--rsa-key-size` (and other options) can be passed by end users of the role.

**Comment by simonspa** [comment]
Still relevant IMHO and would be a nice addition.

**Comment by simonspa** [comment]
@geerlingguy it's very unfortunate this PR was closed, especially since all your comments have been taken into account already by @MicroJoe and the functionality would present a very useful addition to the role.

**Comment by MicroJoe** [comment]
Rebased on `5.0.1`. Please merge.

**Comment by SamK** [comment]
Hi @gagath, your PR is very useful. Could you please rebase on top of `5.1.0`?

**Comment by theS1LV3R** [comment]
Seems the `certbot_create_extra_args` variable is missing a default value (or definition) in `vars/main.yml, giving the following error:

```
The task includes an option with an undefined variable. The error was: {{ certbot_script }} certonly --{{ certbot_create_method  }} {{ '--hsts' if certbot_hsts else '' }} {{ '--test-cert' if certbot_testmode else '' }} --noninteractive --agree-tos --email {{ cert_item.email | default(certbot_admin_email) }} {{ '--expand' if certbot_expand else '' }} {{ '--webroot-path ' if certbot_create_method == 'webroot'  else '' }} {{ cert_item.webroot | default(certbot_webroot) if certbot_create_method == 'webroot' else '' }} {{ certbot_create_extra_args }} -d {{ cert_item.domains | join(',') }} {{ '--pre-hook /etc/letsencrypt/renewal-hooks/pre/stop_services'
  if certbot_create_standalone_stop_services and certbot_create_method == 'standalone'
else '' }} {{ '--post-hook /etc/letsencrypt/renewal-hooks/post/start_services'
  if certbot_create_standalone_stop_services and certbot_create_method == 'standalone'
else '' }}: 'certbot_create_extra_args' is undefined. 'certbot_create_extra_args' is undefined. {{ certbot_script }} certonly --{{ certbot_create_method  }} {{ '--hsts' if certbot_hsts else '' }} {{ '--test-cert' if certbot_testmode else '' }} --noninteractive --agree-tos --email {{ cert_item.email | default(certbot_admin_email) }} {{ '--expand' if certbot_expand else '' }} {{ '--webroot-path ' if certbot_create_method == 'webroot'  else '' }} {{ cert_item.webroot | default(certbot_webroot) if certbot_create_method == 'webroot' else '' }} {{ certbot_create_extra_args }} -d {{ cert_item.domains | join(',') }} {{ '--pre-hook /etc/letsencrypt/renewal-hooks/pre/stop_services'
  if certbot_create_standalone_stop_services and certbot_create_method == 'standalone'
else '' }} {{ '--post-hook /etc/letsencrypt/renewal-hooks/post/start_services'
  if certbot_create_standalone_stop_services and certbot_create_method == 'standalone'
else '' }}: 'certbot_create_extra_args' is undefined. 'certbot_create_extra_args' is undefined

The error appears to be in '/root/.ansible/roles/geerlingguy.certbot/tasks/create-cert-standalone.yml': line 35, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Generate new certificate if one doesn't exist.
  ^ here
```

**Pull request merged**